src/wallet/wallet.h

```diff
@@ -337,6 +335,7 @@ class CWalletTx : public CMerkleTx
     mutable CAmount nImmatureWatchCreditCached;
     mutable CAmount nAvailableWatchCreditCached;
     mutable CAmount nChangeCached;
+    mutable bool fInMempool;
```

Comment:
Instead of `fInMempool` for each wallet transaction, maybe a `std::set<WalletTx*>` with the mempool transactions takes less memory?